plugin/trino-teradata/src/test/java/io/trino/plugin/integration/TeradataJdbcConnectorTest.java

```diff
@@ -0,0 +1,942 @@
+        // Register this specific instance for this test class
+        return TeradataQueryRunner.builder(database).setInitialTables(REQUIRED_TPCH_TABLES).build();
+    }
+
+    @AfterAll
+    public void cleanupTestDatabase()
+    {
+        database.close();
+    }
+
+    @Override
+    protected OptionalInt maxSchemaNameLength()
+    {
+        return OptionalInt.of(TERADATA_OBJECT_NAME_LIMIT);
+    }
+
+    @Override
+    protected void verifySchemaNameLengthFailurePermissible(Throwable e)
+    {
+        assertThat(e).hasMessage(format("Schema name must be shorter than or equal to '%s' characters but got '%s'", TERADATA_OBJECT_NAME_LIMIT, TERADATA_OBJECT_NAME_LIMIT + 1));
+    }
+
+    @Override
+    protected OptionalInt maxColumnNameLength()
+    {
+        return OptionalInt.of(TERADATA_OBJECT_NAME_LIMIT);
+    }
+
+    @Override
+    protected void verifyColumnNameLengthFailurePermissible(Throwable e)
+    {
+        assertThat(e).hasMessageMatching(format("Column name must be shorter than or equal to '%s' characters but got '%s': '.*'", TERADATA_OBJECT_NAME_LIMIT, TERADATA_OBJECT_NAME_LIMIT + 1));
+    }
+
+    @Override
+    @Test
+    public void testDataMappingSmokeTest()
+    {
+        // Skipping the Data Mapping smoke test as this is consuming more time to complete all data type mapping on Teradata ClearScape instance. Will enable once we fix timeout error when running tests on Teradata ClearScape instance
+        skipTestUnless(false);
+    }
+
+    @Override
+    protected OptionalInt maxTableNameLength()
+    {
+        return OptionalInt.of(TERADATA_OBJECT_NAME_LIMIT);
+    }
+
+    @Override
+    protected void verifyTableNameLengthFailurePermissible(Throwable e)
+    {
+        assertThat(e).hasMessageMatching(format("Table name must be shorter than or equal to '%s' characters but got '%s'", TERADATA_OBJECT_NAME_LIMIT, TERADATA_OBJECT_NAME_LIMIT + 1));
+    }
+
+    @Override
+    @Test
+    public void testDistinctLimit()
+    {
+        assertQuery("SELECT COUNT(*) FROM (SELECT DISTINCT orderstatus, custkey FROM orders LIMIT 10)");
+        assertQuery("SELECT DISTINCT custkey, orderstatus FROM orders WHERE custkey = 1268 LIMIT 2");
+        assertQuery("SELECT DISTINCT x " +
+                        "FROM (VALUES 1) t(x) JOIN (VALUES 10, 20) u(a) ON t.x < u.a " +
+                        "LIMIT 100",
+                "SELECT 1");
+    }
+
+    /* Overriding the method as Teradata avg calculations are slightly different than trino so Skipping the results check for avg
+        Expecting actual: (111.660, 111728394.9938271616, 1.117283945E8, 111.6605) to contain exactly in any order: [(111.661, 111728394.9938271605, 1.117283945E8, 111.6605)] */
+    @Override
+    @Test
+    public void testNumericAggregationPushdown()
+    {
+        if (!this.hasBehavior(TestingConnectorBehavior.SUPPORTS_AGGREGATION_PUSHDOWN)) {
```

Comment:
We shouldn't use `hasBehavior` here because each connector already knows its implementation.